### DISCUSSION TITLE
Warning events esx.problem.net.vmknic.ip.duplicate

### DISCUSSION THREAD

**Discussion by adisai123** [vmware/govmomi#2793] [Community Help]
How to catch this event , does govmomi has this event type . I dont see it in github.com/vmware/govmomi/vim25/types

**Comment by embano1** [comment]
How did you get this warning? Can you please show how to reproduce here?

**Reply by adisai123** [reply]
I saw under  monitor/event section of vSphere client.

**Reply by adisai123** [reply]
basically if you assign already used/assigned  ip to interface any interface, you will get this warning event.

**Reply by adisai123** [reply]
my only concern is github.com/vmware/govmomi/vim25/types doesn’t match all vCenter events listed below:
https://www.virten.net/vmware/vcenter-events/

**Reply by embano1** [reply]
Can you please paste a screenshot of the vSphere Client where you see the Fault? 

Btw: Here‘s the `Event` base class and inherited classes: https://vdc-download.vmware.com/vmwb-repository/dcr-public/b50dcbbf-051d-4204-a3e7-e1b618c1e384/538cf2ec-b34f-4bae-a332-3820ef9e7773/vim.event.Event.html

**Reply by embano1** [reply]
I guess you‘re interested in this event then? https://vdc-download.vmware.com/vmwb-repository/dcr-public/b50dcbbf-051d-4204-a3e7-e1b618c1e384/538cf2ec-b34f-4bae-a332-3820ef9e7773/vim.event.DuplicateIpDetectedEvent.html

**Reply by embano1** [reply]
btw: you might want to take a look at https://vmweventbroker.io/ which uses govmomi to react to vSphere events, e.g. by only writing your business logic as a function.
